### ISSUE TITLE
Questions on training protocol and evaluation fairness in transient tasks

### ISSUE THREAD

**Issue by shangguanhaonan** [AI4Science-WestlakeU/RealPDEBench#5]
# Questions on Benchmark Design: Time Handling, Training Protocol, and Evaluation Fairness

Hi, thanks for this great benchmark! After carefully reading the code, I have a few questions about the design choices. Happy to be corrected if I've misunderstood anything.

---

## 1. Inconsistent time dimension handling across models

Different models treat the time axis in fundamentally different ways:

| Model | Time strategy |
|-------|--------------|
| FNO / MWT | 3D spectral transform over `(T, H, W)`; time folded into output channels |
| U-Net | 3D convolution `(B, C, T, H, W)`; time not downsampled |
| WDNO | Input + target **concatenated** then jointly processed by 3D wavelet diffusion |
| DeepONet | Time as a query coordinate in the trunk net |
| Transolver | Time as the depth dimension `D=20`; no explicit output time multiplier |
| DPOT | Built-in sliding-window multi-step prediction |

**Question:** Is this heterogeneity intentional (i.e., "let each model use its natural inductive bias"), or is it considered a confound in the comparison? It seems difficult to attribute performance differences to model quality vs. time-handling strategy.

---

## 2. Training always uses 20→20, regardless of `N_autoregressive` in config

In `train.py`, the dataset is constructed **without** passing `N_autoregressive`:

```python
train_dataset = DatasetClass(
    dataset_name=args.dataset_name,
    dataset_root=args.dataset_root,
    mode='train',
    dataset_type=args.train_data_type,
    mask_prob=args.mask_prob,
    noise_scale=args.noise_scale,
    # N_autoregressive is never passed here
)
```

This means all models always train on the hardcoded default (`N_autoregressive=1`, i.e., 20 input frames → 20 target frames), regardless of what `N_autoregressive` is set to in the config. The in-training validation loop (`pred = model(input)`, line 361) is also a single forward pass with no rollout.

**Question:** Is it intentional that `N_autoregressive` in the config only affects `eval.py` and has zero effect on training? If so, why are different values set per model in the config — is it purely an evaluation design choice?

---

## 3. Train–eval distribution shift for transient tasks

Since training is always single-step (20→20) but `eval.py` calls models autoregressively (`N_autoregressive` times), models with `N_autoregressive > 1` face a distribution shift at test time: their input during rollout is the model's own (potentially erroneous) prediction, not ground-truth data.

For a transient problem like cylinder flow (Kármán vortex street), this error compounding seems non-trivial. 

**Question:** Is training with rollout (e.g., scheduled sampling or multi-step loss) considered out of scope for this benchmark, or is there a plan to add it? It would also help to clarify whether the reported numbers reflect single-step or rollout evaluation.

---

## 4. `N_autoregressive` varies widely across models in eval (1 to 10)

In `eval.py`, different models are evaluated on very different time horizons:

| Models | `N_autoregressive` | Total predicted steps |
|--------|:-----------------:|:--------------------:|
| FNO, DeepONet | 10 | 200 |
| U-Net, WDNO | 5 | 100 |
| CNO, MWT, Transolver | 3 | 60 |
| GalerkinTransformer, DPOT, DMD | 1 | 20 |

FNO must remain accurate after 10 rollout steps; DPOT is evaluated after just 1. **Is comparing their metrics on the same leaderboard meaningful?** If the intent is to evaluate each model at its "natural" prediction length, it would help to document this explicitly so readers don't misinterpret the numbers as apples-to-apples comparisons.

**Comment by Peiyannn** [comment]
Thank you for your interest in our work. I will address your questions one by one below.

1. Our treatment of the temporal dimension basically follows the original open-source implementation of each model, or extends upon it when necessary. We did not intentionally modify how different models handle temporal inputs.

2. Yes, during training, all models are trained only for 20->20 prediction. We evaluate their multi-step autoregressive prediction accuracy only at test time. The `N_autoregressive` parameter in the config file can be freely set according to the user's testing needs.

3. We currently do not plan to add autoregressive training to the codebase, but it should be straightforward to implement based on the existing code. The testing setup corresponding to each table in the paper is specified in the original text.

4. As mentioned above, the current value of `N_autoregressive` in the config is only an example, and users can set it arbitrarily. All comparisons in the paper are fair, with the same `N_autoregressive` used for the compared models. Please refer to the paper for details.

**Comment by shangguanhaonan** [comment]
Thank you for the clarification.
